### ISSUE TITLE
Does git backup support other branches?

### ISSUE THREAD

**Issue by cardosocristian** [opnsense/plugins#3132]
**Important notices**
Our forum is located at https://forum.opnsense.org , please consider joining discussions there in stead of using GitHub for these matters.

Before you ask a new question, we ask you kindly to acknowledge the following:

- [x ] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [ x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.

I'm running opnsense 13.1-RELEASE-p2 stable/22.7-n250229-c8a7fd0d5dc SMP amd64 and I'm trying to run git-backup on a branch other than master, I noticed that apparently it only accepts the master branch in the configuration, because regardless of the branch that is configured, it always tries to run git from a local branch master

```
git-backup unknown error, check log for details (To ssh://gitlab.domain.local/mygroup/backup-opnsense.git ! [rejected] master -> branch2
```

**Comment by AdSchellevis** [comment]
usually you should just choose an non-existing remote branch and it should create one for you. (https://docs.opnsense.org/manual/git-backup.html#conflict-resolution)

**Comment by tobiasehlert** [comment]
I do experience basically the same here.. and here is my finding so far.

In my OPNsense backup setting it doesn't matter what I am writing in the `branch` field, the error in the general log says still `master`.

```plain
2023-02-24T22:42:04 | Notice | configctl | event @ 1677274923.86 exec: system event config_changed
2023-02-24T22:42:04 | Notice | configctl | event @ 1677274923.86 msg: Feb 24 22:42:03 opnsense-prod.my-domain.se config[58341]: [2023-02-24T22:42:03+01:00][INFO] config-event: new_config /conf/backup/config-1677274923.8463.xml
2023-02-24T22:42:04 | Error | php-cgi | git-backup unknown error, check log for details (error: src refspec master does not match any error: failed to push some refs to 'ssh://github.com/my-domain/opnsense-config-prod.git' )
```

But when I navigate in the shell with root user to folder `/conf/backup/git` and push manually to git (with `git push origin main`), all not yet published git history gets into github.

One weird thing here is also that everytime I change settings, the git config for upstream branch is lost.\
So I set the upstream once again and then it works with only `git push`. This is what I do:\
```console
git push --set-upstream origin main
```

I also checked for my configured branches with `git branch` and it says only `main` which is probably right.

```shell
root@opnsense-prod:/conf/backup/git # git branch
* main
```

And when checking in the `.git/config`inside of the git folder, it seems to be generated everytime pressing save in the gui.\
Basically what would be missing there is following part (when using _main_ as branch):
```yaml
[branch "main"]
  remote = origin
  merge = refs/heads/main
```

There seems to be an issue as well regarding the same thing I guess.. #2994

But yeah.. there are multiple questions but I don't know where to look further 😛

**Comment by AdSchellevis** [comment]
The config file should look like:

```
 # cat /conf/backup/git/.git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	sshCommand = ssh -i /conf/backup/git/identity -o StrictHostKeyChecking=accept-new -o PasswordAuthentication=no
[remote "origin"]
	url = ssh://git@github.com/myremote/myrepo.git
	fetch = +refs/heads/*:refs/remotes/origin/*
```

The local copy will always be master, it just pushes to another remote branch, which it should create for itself (or be "bare") in order to be responsible for it's contents.  As explained in https://docs.opnsense.org/manual/git-backup.html

Just checked with github from my end to be sure nothing has changed, but all seems to be in its proper state.

**Comment by AdSchellevis** [comment]
contributed here https://github.com/opnsense/docs/commit/c4bb885a2120135d6e523818347c54ec42eb5d3f, but has nothing todo with the plugin (the plugin doesn't care what the name is, locally it's just always master, remote it can be anything as long as created empty)

code in question:

https://github.com/opnsense/plugins/blob/4546aeb47d62010ea4f8d3e4204b26285be864fe/sysutils/git-backup/src/opnsense/mvc/app/library/OPNsense/Backup/Git.php#L165

**Comment by AdSchellevis** [comment]
corrected the docs in https://github.com/opnsense/docs/commit/46d56a8e71c2d1e1081c68d1407d9a216749cb3e and added a tip about how to enforce a blank branch.
